app/views/rewards/index.html.erb

```diff
@@ -50,7 +50,11 @@
 
           <figcaption class="block-footer text-end">
             <%= link_to 'Show', reward_path(reward), class: 'btn btn-outline-primary' %>
-            <%= link_to 'Buy', new_order_path(reward: reward), class: 'btn btn-outline-success' %>
+            <% if reward.available_for_purchase? %>
```

Comment:
One of the acceptance criteria says "A reward with an online delivery method that has 0 available online codes shouldn't be visible to the user"

I'm not too familiar with the entire app, but I would assume that it means that `Show` is also restricted.

A second point - controller actions should also restrict access if this condition is not met. So `new_order_path(reward: reward)` should reject request (or redirect somewhere) unless `reward.available_for_purchase?`. 

Please let me know if I'm not getting the context correctly.

Reply:
Honestly, I wasn't sure how to understand this ac. Your argument is right so I implemented restrictions for `Show` action when reward is not available for purchase. These restrictions were added to index.html.erb file and also to rewards_controller.rb :)

About second point, CreateOrderService first checks if reward is for sale (it uses `available_for_purchase?` method. If reward is not then it return false from entire `call` method and print error _"Reward is not available now."_ Please let me know if it's enough :) Currently the check looks:
`def reward_for_sale?
    return true if @reward.available_for_purchase?

    @errors << 'Reward is not available now.'
    false
  end`

Reply:
I think that in this shape of the project the service logic is enough.
Thank you for the updates!